test_evm_eth_compliance/src/statetest/runner.rs

```diff
@@ -0,0 +1,377 @@
+                )
+            );
+
+            assert!(
+                info.code.first() != Some(&0xEF),
+                "EIP-3541: Contract code starting with the 0xEF byte is disallowed.",
+            );
+
+            let create_result = test_vm
+                .apply_message(
+                    account,
+                    EAM_ACTOR_ADDR,
+                    TokenAmount::zero(),
+                    fil_actor_eam::Method::CreateAccount as u64,
+                    Some(fil_actor_eam::InitAccountParams {
+                        eth_address: EthAddress(address.as_bytes().try_into().unwrap()),
+                    }),
+                )
+                .unwrap();
+
+            assert!(
+                create_result.code.is_success(),
+                "failed to create the new actor :: {:#?} | Path :: {:#?}",
+                create_result.message,
+                path,
+            );
+
+            let create_return: fil_actor_eam::Create2Return =
+                create_result.ret.deserialize().expect("failed to decode results");
+
+            let actor_address = Address::new_id(create_return.actor_id);
+
+            test_vm.mutate_state(actor_address, |st: &mut EvmState| {
+                let hasher = Code::try_from(SupportedHashes::Keccak256 as u64).unwrap(); // supported hashes are all implemented in multihash
+                let code_hash = multihash::Multihash::wrap(
+                    SupportedHashes::Keccak256 as u64,
+                    &hasher.digest(&info.code).to_bytes(),
+                )
+                .expect("failed to hash bytecode with keccak");
+
+                let cid = store
+                    .put(Code::Blake2b256, &Block::new(IPLD_RAW, info.code.clone()))
+                    .expect("failed to write bytecode");
+
+                let mut slots = StateKamt::new_with_config(store.clone(), KAMT_CONFIG.clone());
+
+                st.bytecode = cid;
+                st.bytecode_hash = code_hash;
+                st.contract_state = slots.flush().expect("failed to flush contract state");
+                st.nonce = info.nonce;
+            });
+
+            pre_contract_cache.insert(*address, actor_address);
+        }
+
+        // Process the "Post" & "transaction" block
+        for (spec_name, tests) in &unit.post {
+            for (id, test) in tests.iter().enumerate() {
+                if skip_post_test(name.as_ref(), spec_name, id + 1) {
+                    continue;
+                }
+
+                info!("Executing TestCase {:#?}::{:#?}::{:#?}", name, spec_name, id,);
+
+                let gas_limit = *unit.transaction.gas_limit.get(test.indexes.gas).unwrap();
+                let gas_limit = u64::try_from(gas_limit).unwrap_or(u64::MAX);
+                let _tx_gas_limit = gas_limit;
+                let tx_data = unit.transaction.data.get(test.indexes.data).unwrap().clone();
+                let _tx_value = *unit.transaction.value.get(test.indexes.value).unwrap();
```

Comment:
Unfortunately our gas model does not translate to the Eth gas model. But it doesn't matter much since I don't think you can perform any useful gas-related asserts here since the TestVM does not account for gas.